metricbeat/docs/modules/googlecloud.asciidoc

```diff
@@ -0,0 +1,95 @@
+* *project_id*: A single string with your GCP Project ID
+* *credentials_file_path*: A single string pointing to the JSON file path reachable by Metricbeat that you have created using IAM.
+* *excluse_labels*: (`true`/`false` default `false`) Do not extract extra labels and metadata information from Metricsets and fetch metrics onlly. At the moment, *labels and metadata extraction is only supported* in Compute Metricset.
+
+[float]
+== Authentication, authorization and permissions.
+Authentication and authorization in Google Cloud Platform can be achieved in many ways. For the current version of the Google Cloud Platform module for Metricbeat, the only supported method is using Service Account JSON files. A typical JSON with a private key looks like this:
+
+```json
+{
+  "type": "service_account",
+  "project_id": "your-project-id",
+  "private_key_id": "a_private_key_id",
+  "private_key": "-----BEGIN PRIVATE KEY-----your private key\n-----END PRIVATE KEY-----\n",
+  "client_email": "some-email@your-project-id.iam.gserviceaccount.com",
+  "client_id": "123456",
+  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+  "token_uri": "https://oauth2.googleapis.com/token",
+  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/metricbeat-testing%40your-project-id.iam.gserviceaccount.com"
+}
+```
+
+Generally, you have to create a Service Account and assign it the following roles or the permissions described on each role:
+
+* `Compute Viewer` (1 permission in use of 138):
+- `monitoring.metricDescriptors.list`
+- `monitoring.timeSeries.list`
+
+* `Monitoring Viewer` (2 permissions in use of 13)
+- `compute.instances.get`
+
+You can play in IAM pretty much with your service accounts and Instance level access to your resources (for example, allowing that everything running in an Instance is authorized to use the Compute API). The module uses Google Cloud Platform libraries for authentication so many possibilities are open but the Module is only supported by using the method mentioned above.
+
+[float]
+== Google Cloud Platform module: Under the hood
+
+Google Cloud Platform offers the https://cloud.google.com/monitoring/api/metrics_gcp[Stackdriver Monitoring API] to fetch metrics from its services. *Those metrics are retrieved one by one*.
+
+If you also want to *extract service metadata and labels* (by setting `exclude_labels` to false, which is the default state). You also make a new API check on the corresponding service. Service labels requires a new API call to extract those metrics. In the worst case the number of API calls will be doubled. In the best case, all metrics come from the same GCP entity and 100% of the required information is included in the first API call (which is cached for subsequent calls).
+
+[float]
+=== Rough estimation of the number of API Calls
```

Comment:
I added a short section about periods but I think it's better to not mention many things that appear directly (and better explained) in GCP docs because we risk to get our docs outdated. I feel like it's safe to assume that if they are worried about how GCP charges them, they'll check GCP and the "under the hood" section will gives them the overall idea of how it might impact them